pkg/policies/security/security_test.go

```diff
@@ -0,0 +1,167 @@
+// Copyright 2021 Allstar Authors
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+
+//     http://www.apache.org/licenses/LICENSE-2.0
+
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package security
+
+import (
+	"context"
+	"encoding/base64"
+	"errors"
+	"net/http"
+	"reflect"
+	"testing"
+
+	"github.com/google/go-github/v32/github"
+	"github.com/ossf/allstar/pkg/config"
+	"github.com/ossf/allstar/pkg/policydef"
+)
+
+var getContents func(context.Context, string, string, string,
+	*github.RepositoryContentGetOptions) (*github.RepositoryContent,
+	[]*github.RepositoryContent, *github.Response, error)
+
+type mockRepos struct{}
```

Comment:
Update to:

```
type mockRepos struct {
  ctx context.Context
  owner, repo, path string
  opts *github.RepositoryContentGetOptions
}

func (m mockRepos) GetContents(...) (...) {
  // Logic operating on m.ctx, m.owner, m.repo etc.
}
```

And then, you just initialize a new instance of mockRepos in the test: 

```
m := mockRepos {
 ctx: ...,
 repo: ...,
}
```